libs/util/nestjs/tracing/src/lib/tracing.module.ts

```diff
@@ -0,0 +1,115 @@
+import { DynamicModule, Module } from "@nestjs/common";
+import {
+  OpenTelemetryModule,
+  OpenTelemetryModuleAsyncOption,
+  ControllerInjector,
+  EventEmitterInjector,
+  GuardInjector,
+  PipeInjector,
+  ScheduleInjector,
+} from "@overbit/opentelemetry-nestjs";
+import { BatchSpanProcessor } from "@opentelemetry/sdk-trace-base";
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-grpc";
+import type { OpenTelemetryModuleConfig } from "@overbit/opentelemetry-nestjs/dist/OpenTelemetryModuleConfig.interface";
+import { AWSXRayPropagator } from "@opentelemetry/propagator-aws-xray";
+import { AWSXRayIdGenerator } from "@opentelemetry/id-generator-aws-xray";
+import { AwsInstrumentation } from "@opentelemetry/instrumentation-aws-sdk";
+import { KafkaJsInstrumentation } from "opentelemetry-instrumentation-kafkajs";
+import { CompositePropagator } from "@opentelemetry/core";
+import { PrismaInstrumentation } from "@prisma/instrumentation";
+
+@Module({
+  imports: [],
+  controllers: [],
+  providers: [],
+  exports: [],
+})
+export class TracingModule extends OpenTelemetryModule {
+  private static addAwsXRayConfiguration(
+    configuration?: Partial<OpenTelemetryModuleConfig>
+  ): Partial<OpenTelemetryModuleConfig> {
+    configuration.instrumentations = configuration.instrumentations || [];
+    configuration.instrumentations.push(
+      new AwsInstrumentation({
+        suppressInternalInstrumentation: true,
+        sqsExtractContextPropagationFromPayload: true,
+      })
+    );
+
+    const traceExporter = new OTLPTraceExporter();
+
+    return {
+      ...configuration,
+      textMapPropagator: new CompositePropagator({
+        propagators: [new AWSXRayPropagator()],
+      }),
+      spanProcessor: new BatchSpanProcessor(traceExporter),
+      traceExporter,
+      idGenerator: new AWSXRayIdGenerator(),
+    };
+  }
+
+  private static addLocalConfiguration(
+    configuration?: Partial<OpenTelemetryModuleConfig>
+  ): Partial<OpenTelemetryModuleConfig> {
+    const collectorOptions = {
+      // url is optional and can be omitted - default is http://localhost:4318/v1/traces,
+      // url: '<opentelemetry-collector-url>',
+    };
+    const traceExporter = new OTLPTraceExporter(collectorOptions);
+
+    return {
+      ...configuration,
```

Comment:
if `configuration` is undefined (since the input is `configuration?`) , then `...configurations` would throw an exception I believe

Reply:
fixed